src/autoscaler/eventgenerator/generator/evaluator.go

```diff
@@ -70,6 +70,10 @@ func (e *Evaluator) doEvaluate(triggerArray []*model.Trigger) {
 			return
 		}
 		for _, appMetric := range appMetricList {
+			if appMetric.Value < 0 {
+				e.logger.Info("should not send trigger alarm to scaling engine because there is nil-value metric", lager.Data{"trigger": trigger, "appMetric": appMetric})
```

Comment:
When aggregator get a empty list from metric-collector, it may mean that the application for aggregating may have some problems that the metric-collector can not retrieve its metrics in the last statWindow. A nil record should be saved so that when do trigger evaluation, if there is a nil record, no alarm should be sent. Above I set the nil record's value to -1, so here it check whether the value is LT 0.